x-pack/plugins/lens/public/datasources/common/field_item.test.tsx

```diff
@@ -5,42 +5,104 @@
-import { render, screen } from '@testing-library/react';
+import { loadFieldStats } from '@kbn/unified-field-list/src/services/field_stats';
+import { fireEvent, render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { EuiLoadingSpinner, EuiPopover } from '@elastic/eui';
 import { InnerFieldItem, FieldItemIndexPatternFieldProps } from './field_item';
 import { coreMock } from '@kbn/core/public/mocks';
-import { mountWithIntl } from '@kbn/test-jest-helpers';
 import { fieldFormatsServiceMock } from '@kbn/field-formats-plugin/public/mocks';
 import { IndexPattern } from '../../types';
 import { chartPluginMock } from '@kbn/charts-plugin/public/mocks';
 import { documentField } from '../form_based/document_field';
+import { KibanaRenderContextProvider } from '@kbn/react-kibana-context-render';
 
 jest.mock('@kbn/unified-field-list/src/services/field_stats', () => ({
   loadFieldStats: jest.fn().mockResolvedValue({}),
 }));
 
-const clickField = async (wrapper: ReactWrapper, field?: string) => {
-  await act(async () => {
-    await wrapper
-      .find(`[data-test-subj="lnsFieldListPanelField-${field}"] .kbnFieldButton__button`)
-      .simulate('click');
-  });
-};
+const corePluginMock = coreMock.createStart();
+
+const indexPattern = {
+  id: '1',
+  title: 'my-fake-index-pattern',
+  timeFieldName: 'timestamp',
+  fields: [
+    {
+      name: 'timestamp',
+      displayName: 'timestamp',
+      type: 'date',
+      aggregatable: true,
+      searchable: true,
+    },
+    {
+      name: 'bytes',
+      displayName: 'bytes',
+      type: 'number',
+      aggregatable: true,
+      searchable: true,
+    },
+    {
+      name: 'memory',
+      displayName: 'memory',
+      type: 'number',
+      aggregatable: true,
+      searchable: true,
+    },
+    {
+      name: 'unsupported',
+      displayName: 'unsupported',
+      type: 'geo',
+      aggregatable: true,
+      searchable: true,
+    },
+    {
+      name: 'source',
+      displayName: 'source',
+      type: 'string',
+      aggregatable: true,
+      searchable: true,
+    },
+    {
+      name: 'ip_range',
+      displayName: 'ip_range',
+      type: 'ip_range',
+      aggregatable: true,
+      searchable: true,
+    },
+    {
+      name: 'geo.coordinates',
+      displayName: 'geo.coordinates',
+      type: 'geo_shape',
+      aggregatable: true,
+      searchable: true,
+    },
+    documentField,
+  ],
+  isTimeBased: jest.fn(),
+} as unknown as IndexPattern;
+
+const dataView = {
+  ...indexPattern,
+  getFormatterForField: jest.fn(() => ({
+    convert: jest.fn((s: unknown) => JSON.stringify(s)),
+  })),
+} as unknown as DataView;
+
+const dataViewsMock = dataViewPluginMocks.createStartContract();
 
 const mockedServices = {
   data: dataPluginMock.createStartContract(),
-  dataViews: dataViewPluginMocks.createStartContract(),
+  dataViews: {
+    ...dataViewsMock,
+    get: jest.fn().mockResolvedValue(dataView),
+  },
```

Comment:
Since `dataViewsMock.get` is already a mock, you could just assign the resolved value directly.

```ts
dataViewsMock.get.mockResolvedValue(dataView);
```

```suggestion
  dataViews: dataViewsMock,
```

Just an idea.

Reply:
Thanks for the suggestion! I like that the syntax is a bit better, but with the first version I like that the code is more colocated where it should be. I'll leave it like this as both have their good sides :)